pkg/controller/iotproject/finalizers.go

```diff
@@ -139,3 +143,18 @@ func cleanupResource(ctx context.Context, c client.Client, project *iotv1alpha1.
 	return reconcile.Result{}, err
 
 }
+
+func cleanupDeviceRegistry(ctx finalizer.DeconstructorContext) (reconcile.Result, error) {
+
+	project, ok := ctx.Object.(*iotv1alpha1.IoTProject)
+
+	if !ok {
+		return reconcile.Result{}, fmt.Errorf("provided wrong object type to finalizer, only supports IoTProject")
+	}
+
+	job := createIotTenantCleanerJob(project.Name)
+
+	result1, err1 := jobsClient.Create(job)
```

Comment:
You need to gracefully handle the fact that the job exists. Also if the job exists, you need to check the status of the job, and let the finalizer be removed if the job succeeded. Otherwise still wait.

When creating a job fails, that would be an error however. Unless it failed to due to "already exists" reasons.